pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -846,6 +810,80 @@ func builderSecretVolume(bbs corev1alpha1.BuildBuilderSpec) corev1.Volume {
 	}
 }
 
+func setupBindingVolumesAndMounts(bindings []ServiceBinding) ([]corev1.Volume, []corev1.VolumeMount) {
+	volumes := make([]corev1.Volume, 0)
+	volumeMounts := make([]corev1.VolumeMount, 0)
+
+	for _, binding := range bindings {
+		switch b := binding.(type) {
+		case *corev1alpha1.ServiceBinding:
+			if b.SecretRef != nil {
+				secretVolume := fmt.Sprintf("service-binding-secret-%s", b.Name)
+				volumes = append(volumes,
+					corev1.Volume{
+						Name: secretVolume,
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: b.SecretRef.Name,
+							},
+						},
+					},
+				)
+				volumeMounts = append(volumeMounts,
+					corev1.VolumeMount{
+						Name:      secretVolume,
+						MountPath: fmt.Sprintf("%s/bindings/%s", platformVolume.MountPath, b.Name),
+						ReadOnly:  true,
+					},
+				)
+			}
+		case *corev1alpha1.CNBServiceBinding:
+			metadataVolume := fmt.Sprintf("binding-metadata-%s", b.Name)
+			volumes = append(volumes,
+				corev1.Volume{
+					Name: metadataVolume,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: *b.MetadataRef,
+						},
+					},
+				},
+			)
+			volumeMounts = append(volumeMounts,
+				corev1.VolumeMount{
+					Name:      metadataVolume,
+					MountPath: fmt.Sprintf("%s/bindings/%s/metadata", platformVolume.MountPath, b.Name),
+					ReadOnly:  true,
+				},
+			)
+			if b.SecretRef != nil {
+				secretVolume := fmt.Sprintf("binding-secret-%s", b.Name)
+				volumes = append(volumes,
+					corev1.Volume{
+						Name: secretVolume,
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: b.SecretRef.Name,
+							},
+						},
+					},
+				)
+				volumeMounts = append(volumeMounts,
+					corev1.VolumeMount{
+						Name:      secretVolume,
+						MountPath: fmt.Sprintf("%s/bindings/%s/secret", platformVolume.MountPath, b.Name),
+						ReadOnly:  true,
+					},
+				)
+			}
+		default:
+			break
```

Comment:
Is this actually an error case?

Reply:
i don't see a case where we would actually ever hit this point, but we can through an error